pkg/czid/validateCSV.go

```diff
@@ -162,8 +163,12 @@ func (c *Client) ValidateSamplesMetadata(projectID int, samplesMetadata SamplesM
 
 	// HACK: new host genomes is a misnomer, all host genomes are returned
 	//   new ones will have ShowAsOption = false
+	// HACK 2: Human always returns ShowAsOption = false
 	hasNewHostGenomes := false
 	for _, hostGenome := range res.NewHostGenomes {
+		if strings.EqualFold(hostGenome.Name, "Human") {
```

Comment:
I'm just at the context of this. Why does `human` get returned in NewHostGenomes?

Reply:
If you look at the response struct for the `POST /metadata/validate_csv_for_new_samples.json` query, one of the json fields returned has the key `newHostGenomes`, but this code only counts a host genome there as "new" if that host genome has `showAsOption: false`. I'm really not sure what's going on there, or why they were named that way, but from what I can tell _all_ host genomes are returned under the `newHostGenomes` key - see https://github.com/chanzuckerberg/czid-web-private/blob/38486cc6bd179894669a8c463731596d022ac683/app/helpers/metadata_helper.rb#L439-L458. So host genomes that are actually new can only be distinguished by the value of the `showAsOption` key.

Why `human` always returns `showAsOption: false` I don't know. I assume it must be a bug in the webapp somewhere but I was unable to track it down.

Reply:
I might be able to shed a little bit of light here from what I learned while adding the Human v2 host genome, but I am also mystified by this idea of "new" host genomes.

The `showAsOption` value comes from the `show_as_option?` method on HostGenome, which is this:
```
def show_as_option?
  user.nil? && deprecation_status.nil?
end
```
Originally, it was just `user.nil?`. The motivation isn't clear to me, but there are some historical breadcrumbs in the comments that seem to indicate that, in the past, users could upload their own host genomes, and thus `user` would get set on the HostGenome. But it was decided we shouldn't show those user-custom HGs as selectable options, only the `user.nil?` no-user HGs should be selectable options since they'd be the ones the CZID team had added on the admin side.

Later on, I added the idea of `deprecation_status` when I rolled out our first v2 of an HG to allow for Human v2. No other HG has a deprecation_status beyond that Human v1, but we could start updating other HGs via this same approach. But I needed to make sure we didn't show "Human" in the list twice -- once for v1 and then again for v2 -- so I expanded it to make sure it only showed the non-deprecated (eg, newest) HG. Which means that it shouldn't be that Human is always returning `showAsOption: false`, but rather that there are two Humans in the list, and one of them is false, while the other is true. If that's not the case, uhhhh, huh. That's weird.

But as for that comment saying "new ones will have ShowAsOption = false", I can't see a way that would be true unless things worked way differently than I thought they did in the past? My guess is it was mistaken, or written in a moment of thrash before the approach settled out. It's true that the endpoint always returned _all_ HGs, it's just that some were marked with showAsOption = false and wouldn't be displayed in FE lists to the user. But it's not that the new HGs came as false, it's that we switched to generally adding HGs on the admin side (like when we service user requests to add a new HG), so they have a `user = nil` and thus get showAsOption = true.

Reply:
Thanks for the clarification, Vincent. After reading what you wrote I have some more thoughts...

If you look at that method in the webapp I linked to above, you'll see that this method does in fact create a new `HostGenome` for that user if `allow_new_host_genomes` is set to true...which it is in this case, since the endpoint the CLI hits invokes `validate_metadata_csv_for_new_samples()`, which calls `validate_metadata_csv_for_samples()` with `allow_new_host_genomes = true`.

Correct me if I'm wrong, but on a second look I think something is seriously wrong here. We're allowing users to create new HostGenomes via the CLI with no limits on how long the name can be other than the size limits of a POST request for Ruby on Rails. For an endpoint that is supposed to just _validate_ the metadata file. This sets off huge red flags for me.

Reply:
Oh my. Yeah, that seems like a bad thing. I don't think it's security-concern level of bad, but I don't really understand how it could possibly provide value in the current state. It seems far more likely to just create a HG that would be useless. Where it potentially creates the new HG in the code you linked, it's
```
hg ||= HostGenome.new(name: name, user: current_user)
```
which would create an HG using almost entirely default values, which I believe devolves to using ERCC indexes. So if a user theoretically did create a new HG for some species we hadn't handled before, it would be pretty much useless, because it would have no connection to that species' bio data, it would just be filtering ERCC and that's it.

You could query the Prod DB and see how many HG rows have been made in the past few years that have a non-NULL `user`. I think any time we create an HG for a user request, user is set to nil for admin-created HGs. I'm not sure how `current_user` works in the above code, but assuming it does what you'd expect, we could figure out if anybody has created an HG through this codepath or not in the last few years. My guess is it's just gone totally unused, but who knows. If it has been unused, I think we should probably remove the ability to arbitrarily create HGs through the CLI, but I'm not sure who has the historical background to guarantee there isn't any kind of still-existing reason to keep this code around.

Reply:
Do we create new HGs for samples uploaded through the web interface?

Reply:
Yes I believe so, if you just write in a random string into the "Host Genome" field, it'll 'create' a new HostGenome object consisting only of ERCC reads. I would _hope_ there is some degree of sanitization of these inputs.

Reply:
To me it sounds like we might need to adjust the `find_by` command [here](https://github.com/chanzuckerberg/czid-web-private/blob/38486cc6bd179894669a8c463731596d022ac683/app/helpers/metadata_helper.rb#L439-L458) to include `deprecation_status`

Reply:
e.g. `HostGenome.find_by(name: "Human", deprecation_status: nil)`

Reply:
I think Ryan's approach would be reasonable.